garrysmod/lua/includes/extensions/entity.lua

```diff
@@ -209,26 +210,57 @@ function meta:InstallDataTable()
 		dt.SetFunc( self, dt.index, value )
 
 	end
+	
+	local function FindUnusedIndex( typename )
+	
+		local tbl = typetable[ typename ]
+		if ( !tbl ) then return 0 end
+		
+		for i=0, 31 do

```

Comment:
replace your loop by this O(1) code
` #tbl + ( next(tbl) and 1 or 0)`

Reply:
This is taking into account there is no gap in the table, it may not apply here

Reply:
I think I missed something and having nil values into your table might be expected, ignore if it's expected to have gaps int he table

Reply:
Yes, I expect gaps.

Reply:
alright, my bad then